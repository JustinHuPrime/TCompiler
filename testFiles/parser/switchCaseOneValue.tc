module foo;

int bar(int i) {
  switch (i) {
    case 'a': { return 0; }
    default: return 1;
  }
}