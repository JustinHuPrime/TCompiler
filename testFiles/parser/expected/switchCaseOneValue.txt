testFiles/parser/input/switchCaseOneValue.tc (code):
FILE(1, 1, STAB(ENTRY(bar, FUNCTION(testFiles/parser/input/switchCaseOneValue.tc, 3, 1, int(int)))), MODULE(1, 1, ID(1, 8, foo, REFERENCES())), FUNDEFN(3, 1, KEYWORDTYPE(3, 1, int), ID(3, 5, bar, REFERENCES(testFiles/parser/input/switchCaseOneValue.tc, 3, 1)), KEYWORDTYPE(3, 9, int), ID(3, 13, i, REFERENCES()), STAB(ENTRY(i, VARIABLE(testFiles/parser/input/switchCaseOneValue.tc, 3, 9, int))), COMPOUNDSTMT(3, 16, STAB(), SWITCHSTMT(4, 3, ID(4, 11, i, REFERENCES(testFiles/parser/input/switchCaseOneValue.tc, 3, 9)), SWITCHCASE(5, 5, LITERAL(5, 10, CHAR('a')), COMPOUNDSTMT(5, 15, STAB(), RETURNSTMT(5, 17, LITERAL(5, 24, UBYTE(0)))), STAB()), SWITCHDEFAULT(6, 5, RETURNSTMT(6, 14, LITERAL(6, 21, UBYTE(1))), STAB())))))
